pkg/cloud/services/networking/port.go

```diff
@@ -289,18 +289,34 @@ func (s *Service) DeletePorts(openStackCluster *infrav1.OpenStackCluster) error
 	return nil
 }
 
-func (s *Service) GarbageCollectErrorInstancesPort(eventObject runtime.Object, instanceName string) error {
-	portList, err := s.client.ListPort(ports.ListOpts{
-		Name: instanceName,
-	})
-	if err != nil {
-		return err
-	}
-	for _, p := range portList {
-		if err := s.DeletePort(eventObject, p.ID); err != nil {
+func (s *Service) GarbageCollectErrorInstancesPort(eventObject runtime.Object, instanceName string, portOpts []infrav1.PortOpts) error {
+	for i := range portOpts {
+		portOpt := &portOpts[i]
+
+		// TODO: whould be nice if gophercloud could be persuaded to accept multiple
+		// names as is allowed by the API in order to reduce API traffic.
+		portList, err := s.client.ListPort(ports.ListOpts{Name: GetPortName(instanceName, portOpt, i)})
+		if err != nil {
 			return err
 		}
+
+		// NOTE: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1476
+		// It is up to the end user to specify a UNIQUE cluster name when provisioning in the
+		// same project, otherwise things will alias and we could delete more than we should.
+		for _, p := range portList {
+			if err := s.DeletePort(eventObject, p.ID); err != nil {
+				return err
+			}
+		}
```

Comment:
We should probably error out if there are multiple ports of that name? Unless we can get to such a case in a single cluster scenario too?

Reply:
Agreed, update incoming.  And now this has been philosophically reviewed, I guess I had better test it!  Luckily I have a 0.8.0 branch... back in a bit.

Reply:
Good news!  It does actually work.